sphericart/src/sphericart_cuda.cpp

```diff
@@ -0,0 +1,188 @@
+#include <iostream>
+#include <stdexcept>
+
+#define _SPHERICART_INTERNAL_IMPLEMENTATION
+
+#include "dynamic_cuda.hpp"
+#include "sphericart.hpp"
+#include "cuda_base.hpp"
+#include "sphericart_cuda.hpp"
+
+using namespace sphericart::cuda;
+
+/*
+    This code checks whether the cuda libraries we'd need to dynamically load are available on the host
+*/
+
+void checkCuda() {
+    if (!CUDA_DRIVER_INSTANCE.loaded()) {
+        throw std::runtime_error(
+            "Failed to load libcuda.so. Try running \"find /usr -name libcuda.so\" and "
+            "appending the directory to your $LD_LIBRARY_PATH environment variable."
+        );
```

Comment:
OK, nice that this is now the only place with an error mesage! Could it be updated to not reference `/usr` since `libcuda` can be elsewhere? This was already done on the other messages but these fell through!

Reply:
I've updated it, do you agree with the new message?

Reply:
yes, looks a lot better!